src/sysinfo.cc

```diff
@@ -75,11 +75,12 @@ bool ReadIntFromFile(const char* file, long* value) {
     char line[1024];
     char* err;
     memset(line, '\0', sizeof(line));
-    CHECK(read(fd, line, sizeof(line) - 1));
-    const long temp_value = strtol(line, &err, 10);
-    if (line[0] != '\0' && (*err == '\n' || *err == '\0')) {
-      *value = temp_value;
-      ret = true;
+    if (read(fd, line, sizeof(line) - 1) >= 0) {
```

Comment:
My point is that it would be nice if the code continued to `CHECK`-fail (i.e., crash the binary) in case of error in debug mode.  That would make it obvious if something goes wrong when reading the file.  (And I agree that a < 0 return value should also `CHECK`-fail.)